### PR TITLE
Feature/orionfix dpdt

### LINKED ISSUE #101: Jobs with 3DEnVar using GDAS ensemble fail on Orion due to memory
## Description
Jobs keep failing on Orion when using GDAS ensemble in 3DEnVar.  The error messages occur in random places due to out of memory related info.

## Steps to reproduce
If someone unfamiliar with the bug follows these steps:
1. submit an analysis job with 3DEnVar using GDAS ensemble.

Potentially relevant details could include:  
-It is likely due to the recent changes in the NOAA RDHPCS system. 

## Output
- Two examples of failure logs:
1.
slurmstepd: error: Detected 1 oom-kill event(s) in step 3495064.0 cgroup. Some of your processes may have been killed by the cgroup out-of-memory handler.
srun: error: Orion-01-01: task 4: Out Of Memory
srun: Terminating job step 3495064.0
slurmstepd: error: *** STEP 3495064.0 ON orion-01-01 CANCELLED AT 2021-10-27T19:35:36 ***

2.
srun: error: Orion-18-69: task 289: Bus error
srun: Terminating job step 3495450.0
slurmstepd: error: *** STEP 3495450.0 ON orion-01-04 CANCELLED AT 2021-10-27T20:22:22 ***

## Status
Bin Liu helped resolve the issue by changing the rocoto/sites/orion.ent file 
from
  <!ENTITY PE_EXTRA        "<native>--partition=orion</native>">
into:
  <!ENTITY PE_EXTRA        "<native>--partition=orion --exclusive</native>">

### LINKED ISSUE #102: Turn on the pressure tendency diagnostic in UFSATM
## Description
Turn on the pressure tendency diagnostic, which is useful to monitor the model instability.

## Proposed solution
Add print_diff_pgr = .true. to the &gfs_physics_nml section of the UFSATM namelist input.nml.

## Related to
- related to hafs-community/hafs/pull/<pr_number>

### PR DESCRIPTION
**Pull request by mrinalbiswas**
## Description of changes
Turn on pressure tendency diagnostic for the forecast
Fix issue #101 so that analysis job uses all the memory resources on the node on Orion

## Issues addressed (optional)
If this PR addresses one or more issues, please provide link(s) to the issue(s) here.
- fixes hafs-community/HAFS/issues/101
- fixes hafs-community/HAFS/issues/102

## Dependencies (optional)
None

## Contributors (optional)
If others worked on this PR besides the author, please include their user names here (using @Mention if possible).

## Tests conducted
@BinLiu-NOAA No testing thus far. What testing you would like to see? 

## Application-level regression test status
Running the HAFS application-level regression tests is currently performed by code reviewers after the developer creates the initial PR. As regression tests are conducted, the testers should use the checklist below to indicate **successful** regression tests. You may add other tests as needed. If a test fails, do not check the box. Instead, describe the failure in the PR comments, noting the platform where the test failed.

- [ ] Jet
- [ ] Hera
- [x] Orion
- [ ] WCOSS Dell P3
- [ ] WCOSS Cray

### CODE REVIEW

**Comment by BinLiu-NOAA** [comment]
@mrinalbiswas, Thanks for making these changes and creating this PR! After the PR #107 being merged, please help to update this PR branch. Then we can do some HAFS application level regression tests. I was thinking of running the RTs on Orion, Jet and Hera might be good enough. Thanks!

File: rocoto/sites/orion.ent [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <!ENTITY ACCOUNT         "&CPU_ACCOUNT;">
   <!ENTITY QUEUE_PE        "batch"> <!-- queue for PE (parallel environment) jobs -->
   <!ENTITY QUEUE_FORECAST  "batch"> <!-- queue for PE (parallel environment) jobs -->
-  <!ENTITY PE_EXTRA        "<native>--partition=orion</native>">
+  <!ENTITY PE_EXTRA        "<native>--partition=orion --exclusive</native>">
```
Comment (BinLiu-NOAA) [inline rocoto/sites/orion.ent:6]: @mrinalbiswas, do we need to make this same change on Jet and Hera, or this is only needed for Orion?

Comment (mrinalbiswas) [reply rocoto/sites/orion.ent:6]: @BinLiu-NOAA I am not sure. I think the issue was reported for Orion, we included only in orion.ent. @evankalina Anyone else reported this issue on other platforms?

Comment (evankalina) [reply rocoto/sites/orion.ent:6]: @mrinalbiswas @BinLiu-NOAA After reviewing the [RDHPCS documentation on memory usage](https://rdhpcs-common-docs.rdhpcs.noaa.gov/wiki/index.php/Running_and_Monitoring_Jobs_on_Jet_and_Hera(Theia)_-_SLURM#How_to_Get_Memory_Usage_Information) for jobs run on Jet/Hera, it is my understanding that a parallel job will automatically be given access to all of the memory on the node(s) it runs on; i.e., there is no need to specify --exclusive, except possibly for serial jobs if they need a lot of memory.

Comment (BinLiu-NOAA) [reply rocoto/sites/orion.ent:6]: @evankalina and @mrinalbiswas, Thanks for checking and confirming! Good to know on Jet and Hera, "--exclusive" is the default. Thanks!

**Comment by BinLiu-NOAA** [comment]
@evankalina, @mrinalbiswas, not sure if the HAFS application regression tests went through on Orion. If so, I think that might be sufficient to merge this PR. Thanks!

**Review by BinLiu-NOAA — APPROVED** [review]
The changes look good to me. Once the HAFS application level regression tests go through on Orion, I think this PR is ready to be merged. Thanks!

**Review by panll — APPROVED** [review]

**Comment by mrinalbiswas** [comment]
@BinLiu-NOAA  RTs passed on Orion

**Comment by BinLiu-NOAA** [comment]
Thanks, @mrinalbiswas and @panll! Ready to merge this PR.

**Pull request merged**